src/github.com/getlantern/balancer/strategy.go

```diff
@@ -0,0 +1,104 @@
+package balancer
+
+import "math/rand"
+
+// Random strategy gives even chance to each dialer, act as a baseline to other
+// strategies.
+func Random(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers: dialers, lessFunc: func(i, j int) bool {
+		// we don't need good randomness, skip seeding
+		if rand.Intn(2) == 0 {
+			return false
+		}
+		return true
+	}}
+}
+
+// Sticky strategy always pick the dialer with largest consecutive success
+// count or the smallest consecutive failure count
+func Sticky(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers: dialers, lessFunc: func(i, j int) bool {
+		q1 := dialers[i].ConsecSuccesses() - dialers[i].ConsecFailures()
+		q2 := dialers[j].ConsecSuccesses() - dialers[j].ConsecFailures()
+		return q1 > q2
+	}}
+}
+
+// Fastest strategy always pick the dialer with lowest recent average connect time
+func Fastest(dialers []*dialer) dialerHeap {
+	return dialerHeap{dialers: dialers, lessFunc: func(i, j int) bool {
+		return dialers[i].AvgDialTime() < dialers[j].AvgDialTime()
```

Comment:
It's indeed a moving average in the form of `(current T + previous AvgT) / 2` though the name doesn't reflect it. If it's not smooth enough in practice we can change it to something like `(current T + (previous AvgT)*9) / 10`. I compared different calculation in `avg_bench_test.go`. Will change the name.